modules/consul-cluster/main.tf

```diff
@@ -157,6 +157,8 @@ module "security_group_rules" {
 # ---------------------------------------------------------------------------------------------------------------------
 
 resource "aws_iam_instance_profile" "instance_profile" {
+  count = "${var.enable_iam_setup}"
+
   name_prefix = "${var.cluster_name}"
   path        = "${var.instance_profile_path}"
   role        = "${aws_iam_role.instance_role.name}"
```

Comment:
This needs the `element(coalescelist(...))` workaround, as `aws_iam_role.instance_role` is now a list

Reply:
Ah yes, well spotted, changing now.